### PR TITLE
Update for terraform-aws-dibbs-ecr-viewer module 0.2.1 :shipit: 

### PR DESCRIPTION
**Pull request by alismx**
# DEVOPS PULL REQUEST

## Related Issue
- https://github.com/CDCgov/terraform-aws-dibbs-ecr-viewer/pull/9
- https://github.com/CDCgov/terraform-aws-dibbs-ecr-viewer/pull/11
- https://github.com/CDCgov/terraform-aws-dibbs-ecr-viewer/pull/12

## Changes Proposed
- Updates to consume changes in the `terraform-aws-dibbs-ecr-viewer` module
- Added AWS ACM certificate data resource.
- Updated the ECS module version from `0.1.2` to `0.2.1`
- Added optional configuration for custom domain and metadata database
- Added new IAM policy actions related to ACM and Secrets Manager

## Additional notes
- Tested with and updated the OIDC module to ensure it continues to work with the latest updates
- Terraform apply for this branch `with internal set to false`: https://github.com/CDCgov/dibbs-aws/actions/runs/11980461986

### CODE REVIEW

File: terraform/implementation/setup/provider.tf [hunk 1]
```diff
@@ -1,17 +0,0 @@
-terraform {
```
Comment (alismx) [inline terraform/implementation/setup/provider.tf:1]: This file was removed in favor of adding it to the existing `backend.tf` file

File: terraform/implementation/setup/backend.tf [hunk 2]
```diff
@@ -3,10 +3,20 @@ terraform {
   required_providers {
     aws = {
       source  = "hashicorp/aws"
-      version = "=5.70.0"
+      version = "~> 5.56.1"
```
Comment (shanice-skylight) [inline terraform/implementation/setup/backend.tf:6]: Did you intentionally go down a version for hashicorp/aws?

**Review by shanice-skylight — COMMENTED** [review]
The version comment is my only question. Everything else is GTG

File: terraform/implementation/setup/backend.tf (hunk 2 above)
Comment (alismx) [reply terraform/implementation/setup/backend.tf:6]: Yes, this was intentional! It was never run with 5.7.

**Review by shanice-skylight — APPROVED** [review]
LGTM

**Pull request merged**